cmd/observe/io_reader_observer.go

```diff
@@ -101,10 +115,42 @@ func (c *ioReaderClient) Recv() (*observer.GetFlowsResponse, error) {
 		if !filters.Apply(c.allow, c.deny, &v1.Event{Timestamp: res.Time, Event: res.GetFlow()}) {
 			continue
 		}
-		return &res, nil
+		switch {
+		case !c.request.GetFirst() && c.request.GetNumber() != 0 && c.request.GetNumber() != uint64Max: // --last
```

Comment:
nit: we can simplify the `--last` condition a bit, since `c.buffer` is initialized only when `--last` is provided.
```suggestion
		case c.buffer != nil: // --last
```